modules/react-map-gl-draw/src/mode-handlers/mode-handler.js

```diff
@@ -0,0 +1,380 @@
+// @flow
+
+import uuid from 'uuid';
+
+import type { MjolnirEvent } from 'mjolnir.js';
+import type WebMercatorViewport from 'viewport-mercator-project';
+import type { Position } from '@nebula.gl/edit-modes';
+import type {
+  Id,
+  ModeHandlerState,
+  ModeHandlerProps,
+  BaseEvent,
+  ClickEvent,
+  DraggingEvent,
+  HoverEvent,
+  PointerMoveEvent,
+  EditAction
+} from '../types';
+
+import { MODE_TO_GEOJSON_TYPE, MODE_TO_RENDER_TYPE, ELEMENT_TYPES, EDIT_TYPES } from '../constants';
+import Feature from '../feature';
+import { getScreenCoords, parseEventElement } from './utils';
+
+export const defaultModeHandlerState: ModeHandlerState = {
+  // hover
+  hoveredFeatureId: null,
+  hoveredLngLat: null,
+  hoveredVertexIndex: -1,
+  hoveredSegmentId: -1,
+
+  // drag
+  draggingVertexIndex: -1,
+  startDragPos: null,
+  isDragging: false,
+  didDrag: false,
+
+  // intermediate mouse position when drawing
+  uncommittedLngLat: null
+};
+
+export default class ModeHandler {
```

Comment:
I think we should rename this to `EditMode` to be consistent with the new naming in the `edit-modes` module. We still have `ModeHandler`, but that will all be refactored to be "edit modes".